### DISCUSSION TITLE
swift error using lldb-vscode | codellb

### DISCUSSION THREAD

**Discussion by alloc33** [mfussenegger/nvim-dap#573] [Q&A]
OS: macOS 12.3.1
M1

In both cases using lldb-vscode or codellb I got this error

```error
Error on launch: Could not create a target for a program '/Users/nshv_a/Repos/eth_swift/': '/Users/nshv_a/Repos/eth_swift' doesn't contain any 'host' platform architectures: arm64e, arm64, x86_64, x86_64, arm64, arm64e.
```

using lldb-vscode:

```lua
local dap = require('dap')
dap.set_log_level('DEBUG')
dap.adapters.lldb = {
	type = 'executable',
	command = '/opt/homebrew/Cellar/llvm/13.0.1_1/bin/lldb-vscode', -- adjust as needed, must be absolute path
	name = 'lldb'
}

dap.configurations.swift = {
	{
		name = 'Launch',
		type = 'lldb',
		request = 'launch',
		program = function()
			return vim.fn.input('Path to executable: ', vim.fn.getcwd() .. '/', 'file')
		end,
		cwd = '${workspaceFolder}',
		stopOnEntry = false,
		args = { "xcrun swift build -c release --arch arm64" },
	},
}
```

Setup for codelldb:

```lua
local dap = require('dap')
dap.adapters.codelldb = {
  type = 'server',
  host = '127.0.0.1',
  port = 13000 -- 💀 Use the port printed out or specified with `--port`
}

local dap = require('dap')
dap.configurations.swift = {
  {
    name = "Launch file",
    type = "codelldb",
    request = "launch",
    program = function()
      return vim.fn.input('Path to executable: ', vim.fn.getcwd() .. '/', 'file')
    end,
    cwd = '${workspaceFolder}',
    stopOnEntry = true,
  },
}
```

LOGS (using lldb-vscode):
[ DEBUG ] 2022-06-03T14:47:59Z+0800 ] ...rvim/site/pack/packer/start/nvim-dap/lua/dap/session.lua:921 ]	"Spawning debug adapter"	{
  command = "/opt/homebrew/Cellar/llvm/13.0.1_1/bin/lldb-vscode",
  name = "lldb",
  type = "executable"
}
[ DEBUG ] 2022-06-03T14:47:59Z+0800 ] ...rvim/site/pack/packer/start/nvim-dap/lua/dap/session.lua:1110 ]	"request"	{
  arguments = {
    adapterID = "nvim-dap",
    clientId = "neovim",
    clientname = "neovim",
    columnsStartAt1 = true,
    linesStartAt1 = true,
    locale = "en_ID.UTF-8",
    pathFormat = "path",
    supportsRunInTerminalRequest = true,
    supportsVariableType = true
  },
  command = "initialize",
  seq = 0,
  type = "request"
}
[ DEBUG ] 2022-06-03T14:47:59Z+0800 ] ...rvim/site/pack/packer/start/nvim-dap/lua/dap/session.lua:756 ]	{
  body = {
    exceptionBreakpointFilters = { {
        default = false,
        filter = "cpp_catch",
        label = "C++ Catch"
      }, {
        default = false,
        filter = "cpp_throw",
        label = "C++ Throw"
      }, {
        default = false,
        filter = "objc_catch",
        label = "Objective C Catch"
      }, {
        default = false,
        filter = "objc_throw",
        label = "Objective C Throw"
      }, {
        default = false,
        filter = "swift_catch",
        label = "Swift Catch"
      }, {
        default = false,
        filter = "swift_throw",
        label = "Swift Throw"
      } },
    supportTerminateDebuggee = true,
    supportsCompletionsRequest = false,
    supportsConditionalBreakpoints = true,
    supportsConfigurationDoneRequest = true,
    supportsDelayedStackTraceLoading = true,
    supportsEvaluateForHovers = true,
    supportsExceptionInfoRequest = true,
    supportsExceptionOptions = true,
    supportsFunctionBreakpoints = true,
    supportsGotoTargetsRequest = false,
    supportsHitConditionalBreakpoints = true,
    supportsLoadedSourcesRequest = false,
    supportsModulesRequest = false,
    supportsProgressReporting = true,
    supportsRestartFrame = false,
    supportsRestartRequest = false,
    supportsRunInTerminalRequest = true,
    supportsSetVariable = true,
    supportsStepBack = false,
    supportsStepInTargetsRequest = false,
    supportsValueFormattingOptions = true
  },
  command = "initialize",
  request_seq = 0,
  seq = 0,
  success = true,
  type = "response"
}
[ DEBUG ] 2022-06-03T14:47:59Z+0800 ] ...rvim/site/pack/packer/start/nvim-dap/lua/dap/session.lua:1110 ]	"request"	{
  arguments = {
    args = { "xcrun swift build -c release --arch arm64" },
    cwd = "/Users/nshv_a/Repos/eth_swift",
    name = "Launch",
    program = "/Users/nshv_a/Repos/eth_swift/",
    request = "launch",
    stopOnEntry = false,
    type = "lldb"
  },
  command = "launch",
  seq = 1,
  type = "request"
}
[ DEBUG ] 2022-06-03T14:47:59Z+0800 ] ...rvim/site/pack/packer/start/nvim-dap/lua/dap/session.lua:756 ]	{
  command = "launch",
  message = "Could not create a target for a program '/Users/nshv_a/Repos/eth_swift/': '/Users/nshv_a/Repos/eth_swift' doesn't contain any 'host' platform architectures: arm64e, arm64, x86_64, x86_64, arm64, arm64e.",
  request_seq = 1,
  seq = 0,
  success = false,
  type = "response"
}
[ INFO ] 2022-06-03T14:47:59Z+0800 ] ...rvim/site/pack/packer/start/nvim-dap/lua/dap/session.lua:938 ]	"Closed all handles"
[ INFO ] 2022-06-03T14:47:59Z+0800 ] ...rvim/site/pack/packer/start/nvim-dap/lua/dap/session.lua:941 ]	"Process closed"	83908	false

Is it even possible to setup swift debugging?

**Comment by mfussenegger** [comment]
Did you see https://github.com/vadimcn/vscode-lldb/wiki/Swift and https://github.com/vadimcn/vscode-lldb/blob/54d4fbb2dead83e98bd44d07057b67efc4305531/MANUAL.md#alternate-lldb-backends ?

**Reply by alloc33** [reply]
> Did you see https://github.com/vadimcn/vscode-lldb/wiki/Swift and https://github.com/vadimcn/vscode-lldb/blob/54d4fbb2dead83e98bd44d07057b67efc4305531/MANUAL.md#alternate-lldb-backends ?

I made it work with LLDB from Xcode, but I have empty terminal (can't use commands like 'po', only possible to debug printing to console)
